src/array/array.ts

```diff
@@ -0,0 +1,56 @@
+import { AlignedType, type Options, type Packed } from "../types/mod.ts";
+
+export class ArrayType<T> extends AlignedType<T[]> implements Packed<T[]> {
+  constructor(readonly type: AlignedType<T>, readonly length: number) {
+    super(type.byteAlignment);
+  }
+
+  readUnaligned(dt: DataView, options: Options = { byteOffset: 0 }): T[] {
+    const result = [];
+    result.length = this.length;
+    result[0] = this.type.readUnaligned(dt, options);
```

Comment:
What if the length of the array is 0? Then this is oob right?

Reply:
Not oob but it def isn't the behavior that we want. This will just insert a element at index 0 without any problem

Reply:
btw is there any usecase for `new ArrayType(T, 0)`? Cuz if not then I think it's best to throw a type error when constructing with a `0` length array

Reply:
I don't think there is any use case, but is it really "wrong"? I can image some kind of imaginary case where an array type is constructed with an arbitrary array length, which can be 0. In that case I believe i would be better to just allow empty, zero-sized, phantom arrays.

Reply:
Sounds rly hacky to do it that way. I feel like its better if the user would do something like this rather than us returning it as empty array.

```ts
const len = u8.read(dt);
if (len === 0) return [];
return new ArrayType(T, len).read(dt);
```

Reply:
`[u8; 0]` in rust is OK, and I mean it does make sense but I also get why it feels useless.

Reply:
I'll add a check for empty items (also `{}` and empty tuple)